modules/effects/src/main/scala/dev/profunktor/redis4cats/runner.scala

```diff
@@ -102,12 +101,13 @@ private[redis4cats] class RunnerPartiallyApplied[F[_]: Concurrent: Log: Parallel
   private def joinOrCancel[H <: HList, G <: HList](ys: H, res: G)(isJoin: Boolean): F[HList] =
     ys match {
       case HNil => Applicative[F].pure(res)
-      case HCons((h: Fiber[F, Any] @unchecked), t) if isJoin =>
-        h.join.flatMap(x => joinOrCancel(t, x :: res)(isJoin))
-      case HCons((h: Fiber[F, Any] @unchecked), t) =>
+      case HCons((h: Fiber[F, Throwable, Any] @unchecked), t) if isJoin =>
+        h.joinWithNever.flatMap(x => joinOrCancel(t, x :: res)(isJoin))
+      case HCons((h: Fiber[F, Throwable, Any] @unchecked), t) =>
         h.cancel.flatMap(x => joinOrCancel(t, x :: res)(isJoin))
       case HCons(h, t) =>
         Log[F].error(s"Unexpected result: ${h.toString}") >> joinOrCancel(t, res)(isJoin)
     }
 
+  private val promiseAlreadyCompleted = new AssertionError("Promise already completed")
```

Comment:
Yep - since `Deferred` become polymorphic in its error type it can't instantiate an error value, so it returns boolean to represent success or failure.